web-app/src/main/java/com/google/sps/data/Analysis.java

```diff
@@ -0,0 +1,56 @@
+package com.google.sps.data;
+
+import java.util.HashMap;
+import java.util.Map.Entry;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Collections;
+
+public class Analysis {
+  private final int numReports;
+  private final List<String> frequentTypes;
+  private HashMap<String, Integer> crimeTypeCountMap;
+
+  private void countCrimeType(List<PoliceReport> reports) {
+    crimeTypeCountMap = new HashMap<String, Integer>();
+
+    for (PoliceReport report : reports) {
+      String crimeType = report.getCrimeType();
+      Integer count = crimeTypeCountMap.get(crimeType);
+      if (count == null) {
+        crimeTypeCountMap.put(crimeType, 1);
+      } else {
+        crimeTypeCountMap.put(crimeType, count + 1);
+      }
+    }
+  }
+
+  private List<String> getTopNTypes(List<PoliceReport> reports, int n) {
+    countCrimeType(reports);
```

Comment:
Should we be passing crimeTypeCountMap back and forth instead of having it global to the class?